fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -131,41 +131,24 @@ public LogicalMetric make(String metricName, String dependentMetric) {
     protected abstract int getDependentMetricsRequired();
 
     /**
-     * Fetch the TemplateDruidQuery of the dependent metric from the Metric Dictionary.
-     *
-     * @param name  Name of the metric to fetch the template druid query from
-     * @return The template druid query of the metric
-     */
-    protected TemplateDruidQuery getDependentQuery(String name) {
-        LogicalMetric dependentMetric = metrics.get(name);
-        return dependentMetric.getTemplateDruidQuery();
-    }
-
-    /**
      * A helper function returning the resulting aggregation set from merging one or more template druid queries.
      *
      * @param names  Names of the metrics to fetch and merge the aggregation clauses from
      *
      * @return The merged query
      */
     protected TemplateDruidQuery getMergedQuery(List<String> names) {
-        if (names.size() <= 0) {
-            String message = "At least 1 name is needed to merge aggregations";
-            LOG.error(message);
-            throw new IllegalArgumentException(message);
-        }
-
-        // Get the initial query
-        TemplateDruidQuery query = getDependentQuery(names.get(0));
-
         // Merge in any additional queries
-        for (int i = 1; i < names.size(); i++) {
-            TemplateDruidQuery additionalQuery = getDependentQuery(names.get(i));
-            query = query.merge(additionalQuery);
-        }
-
-        // Return the merged query
-        return query;
+        return names.stream()
+                .map(metrics::get)
+                .map(LogicalMetric::getTemplateDruidQuery)
+                .reduce(TemplateDruidQuery::merge)
```

Comment:
Does merge handle `null` correctly? I know we're collapsing the error check to the end of this stream (which I'm a fan of), but I just want to make sure that the merge will still be working correctly.